fgpyo/util/logging.py

```diff
@@ -163,6 +164,23 @@ def record_alignment(
         else:
             return self.record(rec.reference_name, rec.reference_start + 1)
 
+    def record_alignments(
+        self,
+        recs: Iterable[AlignedSegment],
```

Comment:
I think this is a sensible and helpful addition to the library.

My only suggestion is to type this more strictly as a `Collection`, so the user does not inadvertently pass a generator which would be consumed by the function.

```suggestion
        recs: Collection[AlignedSegment],
```

Reply:
@msto I do like that idea. What's awkward is almost everything on template returns an iterator even when it doesn't need to (it's just returning things it already has in memory unaltered). So, if I upgrade to `Collection` I'd have to do this:

```python
progress.record_alignments(recs=list(template.all_recs()))
```

Not the worst, but a little awkward.

Reply:
I thought a bit more about this and I left the typing as the most generic thing necessary for the function to work, which is `Iterable`. The function is named "record_alignments" so it should be unsurprising that it will advance/consume/iterate-over whatever you supply it.